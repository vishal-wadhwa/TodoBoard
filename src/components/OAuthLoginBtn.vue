<template>
  <span
    class='v-chip chip-wrapper ma-0'
    :style='{background: btnBg, color: textColor, border: `1px solid ${iconBg}`}'
    :title='title'
  >
    <v-avatar v-if='!!iconSrc' class='ma-0 pa-1' size='32' :style='{background: iconBg}'>
      <img :src='iconSrc' alt='login'>
    </v-avatar>
    &nbsp;{{title}}&nbsp;
  </span>
</template>
<script>
export default {
  props: {
    iconSrc: String,
    title: String,
    iconBg: { type: String, default: 'rgba(0, 0, 0, 0.15)' },
    btnBg: { type: String, default: 'rgba(0, 0, 0, 0.05)' },
    textColor: { type: String, default: 'black' }
  }
}
</script>
<style lang="stylus" scoped>
.chip-wrapper {
  max-width: 34px;
  max-height: 34px;
  overflow: hidden;
  transition: max-width ease 0.5s;
  cursor: pointer;

  &:hover {
    @media screen and (min-width: 600px) {
      max-width: 100%;
    }
  }
}
</style>
