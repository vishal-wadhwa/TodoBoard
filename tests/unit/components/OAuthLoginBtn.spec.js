import { shallowMount, createLocalVue, mount } from '@vue/test-utils'
import OAuthLoginBtn from '@/components/OAuthLoginBtn'
import Vuetify from 'vuetify'
import Vue from 'vue'
Vue.use(Vuetify)

describe('OAuthLoginBtn.vue', () => {
  // required to vuetify [warn] in snapshot test
  // const app = document.createElement('div')
  // app.setAttribute('data-app', true)
  // document.body.appendChild(app)

  let localVue = null
  const factory = (opts = {}, full = false) => {
    return full ? mount(OAuthLoginBtn, opts) : shallowMount(OAuthLoginBtn, opts)
  }
  beforeEach(() => {
    localVue = createLocalVue()
    // localVue.use(Vuetify)
  })

  const title = 'Google'
  const iconSrc = 'some_image.jpg'
  const iconBg = 'red'
  const btnBg = 'green'
  const textColor = 'blue'

  it('does not render icon by default', () => {
    const wrapper = factory({
      localVue
    })
    expect(wrapper.find('vavatar-stub').exists()).toBeFalsy()
  })

  it('renders title passed as prop', () => {
    const wrapper = factory({
      propsData: { title },
      localVue
    })
    expect(wrapper.text()).toMatch(title)
  })

  it('has correct iconSrc', () => {
    const wrapper = factory({
      propsData: { iconSrc },
      localVue
    })
    expect(wrapper.find('img').attributes('src')).toMatch(iconSrc)
  })

  it('should match default snapshot', () => {
    const wrapper = factory({
      localVue
    }, true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should match snapshot with props', () => {
    const wrapper = factory({
      localVue,
      propsData: { title, iconSrc, iconBg, btnBg, textColor }
    }, true)

    expect(wrapper.html()).toMatchSnapshot()
  })
})
