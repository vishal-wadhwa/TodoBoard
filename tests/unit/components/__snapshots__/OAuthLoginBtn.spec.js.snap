// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OAuthLoginBtn.vue should match default snapshot 1`] = `
<span class="v-chip chip-wrapper ma-0" style="background: rgba(0, 0, 0, 0.05); color: black;"><!---->
  &nbsp;&nbsp;
</span>
`;

exports[`OAuthLoginBtn.vue should match snapshot with props 1`] = `
<span title="Google" class="v-chip chip-wrapper ma-0" style="background: green; color: blue; border: 1px solid red;"><div class="v-avatar ma-0 pa-1" style="height: 32px; width: 32px; background: red;"><img src="some_image.jpg" alt="login"></div>
  &nbsp;Google&nbsp;
</span>
`;
